### ISSUE TITLE
Lack of published data fields in Elastic

### ISSUE THREAD

**Issue by amerireza** [ThejanRupasinghe/ei-elastic-custom-publisher#4]
I am using this component and it works well but, I can see so many data when I use ei-analytics but published data from ei-elastic publisher is just a few of them.
These are fields I get when I use ei-analytics.

metaTenantId | messageFlowId | host | hashCode | componentName | componentType | componentIndex | componentId | startTime | endTime | duration | beforePayload | afterPayload | contextPropertyMap | transportPropertyMap | children | entryPoint | entryPointHashcode | faultCount | eventTimestamp

**Comment by ThejanRupasinghe** [comment]
Yes. This component was developed in relation to the requirement explained in [this article](https://wso2.com/library/article/2017/12/monitoring-wso2-enterprise-integrator-logs-and-statistics-with-elastic-stack-elk-stack).
So it only extracts a few fields from the PublishingFlow which are needed to build the [visualizations](https://wso2.com/library/article/2017/12/monitoring-wso2-enterprise-integrator-logs-and-statistics-with-elastic-stack-elk-stack/#visualizing) in Kibana.

It can be improved to extract everything you have mentioned, as it is done here.
https://github.com/ThejanRupasinghe/ei-elastic-custom-publisher/blob/d141e40c4757900df9c8562b280ee1fb8419229b/src/main/java/org/wso2/custom/elastic/publisher/publish/ElasticStatisticsPublisher.java#L62-L107

**Comment by amerireza** [comment]
OK, I wanna add other event types but I don't know how can I see all types. What is the abstract or parent class of these all event types?

**Comment by ThejanRupasinghe** [comment]
[PublishingFlow](https://github.com/wso2/wso2-synapse/blob/9083d4f8a3afcde744cc7df5a320309a2f5b18d2/modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingFlow.java) from [wso2-synapse](https://github.com/wso2/wso2-synapse) is intercepted in the above shown code. It has an array of [PublishingEvents](https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEvent.java), which the message flow has gone through. You can iterate through it and extract the needed fields.

**Comment by amerireza** [comment]
Thank you for the information. But the `componentIndex` variable is private and `eventTimestamp` variables are not existing in the `PublishingEvents.java` class.

**Comment by ThejanRupasinghe** [comment]
Yes. `componentIndex` is private, but can access event related times from [here](https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEvent.java#L106-L128). 
[ElasticMediationFlowObserver](https://github.com/ThejanRupasinghe/ei-elastic-custom-publisher/blob/master/src/main/java/org/wso2/custom/elastic/publisher/observer/ElasticMediationFlowObserver.java) is written using the [MessageFlowObserver](https://github.com/wso2/carbon-mediation/blob/master/components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.das.messageflow.data.publisher/src/main/java/org/wso2/carbon/das/messageflow/data/publisher/observer/MessageFlowObserver.java) interface exposed by the EI Analytics profile, so this is the only visibility we have here.

**Comment by amerireza** [comment]
What do you say, if we add a `getter` for the `componentIndex` in the `PublishingEvent.java` to be able to retrieve the value, in the next versions?

**Comment by ThejanRupasinghe** [comment]
Yes. We with that change we should be able to access `componentIndex` here.

**Comment by amerireza** [comment]
> Yes. We with that change we should be able to access `componentIndex` here.

I have committed my changes [here](https://github.com/wso2/wso2-synapse/pull/1344) and it is waiting for review.
